pucas/ldap.py

```diff
@@ -88,26 +88,22 @@ def user_info_from_ldap(user):
             # multiple attributes OR where it is missing, in which case
             # handle the LDAPCursorError and set None to empty string.
 
-            # if the user passed a list, try in order until we get the attr
-            # or we fail to do so.
-            if isinstance(ldap_attr, list):
-                for val in ldap_attr:
-                    try:
-                        setattr(user, user_attr, str(getattr(user_info, val)))
-                        break
-                    except LDAPCursorError:
-                        # NOTE: it would be more efficient to do this as a single
-                        # catchall if the attribute ends up unset, but
-                        # the behavior of Mock __getattr__ (which returns a
-                        # a Mock if undefined), makes it hard to test.
-                        # Opted for testable over minor efficiency gain.
-                        setattr(user, user_attr, '')
-            else:
-            # if the user passed a string, try to set it, catch error
+            # if just a string, convert to a list so handling can be uniform
+            if isinstance(ldap_attr, str):
+                ldap_attr = [ldap_attr]
+
+            # iterate through the list items and break on the first one to
+            # correct set without raising LDAPCursorError
+            for val in ldap_attr:
                 try:
-                    setattr(user, user_attr, str(getattr(user_info, ldap_attr)))
+                    setattr(user, user_attr, str(getattr(user_info, val)))
+                    break
                 except LDAPCursorError:
-                    setattr(user, user_attr, '')
+                    pass
+            # user getattr to check for a still unset value, in which case
```

Comment:
I think this comment could be clearer as to why we need this logic - it seemed to me on first look that we shouldn't override default values on on `User`, but I think I remember that this method can be used to update existing users, so is the scenario that data could have changed and we should clear it out?

Reply:
It's actually just working around `attr_map` a few lines previously using `None` rather than an empty string (which has Django implications when a string field is `blank` but not null). I think all the stock user attributes are stringy in some capacity.

Should we just use:
    `attr_map = settings.PUCAS_LDAP.get('ATTRIBUTE_MAP', '')`

at l. 77 instead or does `None` have special significance here I'm missing.

Reply:
Looking at this again, I realize that what I was doing was handling my uniform conversion to list above. It helps handle LDAP values that happen to be lists (even when they have, in fact, one value). My reasoning here is that mostly we're mapping a string to a string, but LDAP does include list values. This simplification ensures we get *something* if it's there, and in most cases, the exact something we're looking for.